activerecord/lib/active_record/persistence.rb

```diff
@@ -403,6 +409,12 @@ def _delete_record(constraints) # :nodoc:
         connection.delete(dm, "#{self} Destroy")
       end
 
+      def _insert_attributes(attributes)
```

Comment:
This isn't right for upserts. We'd need an `_upsert_attributes` method that only returns `timestamp_attributes_for_update_in_model`.